src/main/java/com/twentyonec/ItemsLogger/listeners/DeathSave.java

```diff
@@ -8,24 +8,27 @@
 
 import com.twentyonec.ItemsLogger.ItemPlayer;
 import com.twentyonec.ItemsLogger.ItemsLogger;
+import com.twentyonec.ItemsLogger.utils.Cause;
+import com.twentyonec.ItemsLogger.utils.Permissions;
+import com.twentyonec.ItemsLogger.utils.Storage;
 
 public class DeathSave implements Listener {
 
 	final private ItemsLogger plugin = ItemsLogger.getPlugin();
+	final private Storage storage = Storage.getStorage(plugin);
 
 	@EventHandler
 	public void onDeathSave(final PlayerDeathEvent event) {
 
 		final Player player = event.getEntity();
-		if (player.hasPermission("itemslogger.log")) {
-			final String cause = event.getDeathMessage();
-			final ItemPlayer itemPlayer = new ItemPlayer(player, cause);
-
-			plugin.debugMessage("Attempting to save player data");
+		if ((plugin.getConfigManager().getDeath()) && 
+				(player.hasPermission(Permissions.PERMISSION_LOG))) {
+			
+			final ItemPlayer itemPlayer = new ItemPlayer(player, Cause.DEATH);
 			new BukkitRunnable() {
 				@Override
 				public void run() {
-					itemPlayer.savePlayer();
+					storage.saveItemPlayer(itemPlayer);
```

Comment:
Always use `this` for class variables
```suggestion
					this.storage.saveItemPlayer(itemPlayer);
```